### PR TITLE
Respect DisabledByDefault for custom cops

### PR DESCRIPTION
**Pull request by XrXr**
Before this commit, DisabledByDefault has no effect on
custom cops and they are enabled when the config files
contain no mention of them. This is a problematic
behavior as CI servers might load multiple custom cop
and run Rubocop over multiple repos. Custom cops end
up enabled for repos that don't care about them, even
though DisabledByDefault is set to true.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/config.rb [hunk 1]
```diff
@@ -710,7 +710,10 @@ def enable_cop?(qualified_cop_name, cop_options)
         end
       end
 
-      cop_options.fetch('Enabled', true)
+      cop_options.fetch('Enabled') do
+        disabled_by_default = for_all_cops['DisabledByDefault']
```
Comment (bbatsov) [inline lib/rubocop/config.rb]: I guess you can simplify this to simply `!for_all_cops['DisabledByDefault']`.

**Comment by bbatsov** [comment]
The changes look good, but your branch has to be rebased on top of the current `master` branch due to merge conflicts.

File: lib/rubocop/config.rb (hunk 1 above)
Comment (XrXr) [reply lib/rubocop/config.rb]: 👍 I opted for the verbose version because it looked nicer.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,7 +9,7 @@
 * Add support for subclassing using `Class.new` to `Lint/InheritException`. ([@houli][])
 * [#6779](https://github.com/rubocop-hq/rubocop/issues/6779): Add new cop `Style/NegativeUnless` that checks for unless with negative condition. ([@tejasbubane][])
 * [#6649](https://github.com/rubocop-hq/rubocop/pull/6649): Layout/AlignHash supports list of options. ([@stoivo][])
-
+* Respect DisabledByDefault for custom cops. ([@XrXr][])
```
Comment (bbatsov) [inline CHANGELOG.md]: This should go in the `master` section of the changelog, under `Bug fixes`.

Comment (XrXr) [reply CHANGELOG.md]: Updated.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**